### PR TITLE
[Data Explorer]: Message, Data, and Transaction Filtering

### LINKED ISSUE #47: Enhanced filtering
The FireFly core API has great filtering options (https://labs.hyperledger.org/firefly/swagger/swagger.html#/default/getMsgs) that the explorer should be leveraging. 

design sketches:
FF-Dark-Messages Timeline - Added Filter
FF-Dark-Messages Timeline Open Filter

These filters should be stored in URL query params, so they can be preserved across links.

### PR DESCRIPTION
**Pull request by shorsher**
This PR adds enhanced filtering for the message, data, and transactions views of the data module.

* Introduces a new modal for selecting the `field`, `operator`, and `value` of the filter
* A display bar to manage your active filters
* Filters stored as query params for easy linking & sharing

closes #47 

Update: adjusted select border color and filter button height
Screen Shot 2021-10-29 at 10 01 29 AM

Screen Shot 2021-10-27 at 11 28 17 PM

Screen Shot 2021-10-27 at 11 29 51 PM

### CODE REVIEW

File: src/core/utils.ts [hunk 1]
```diff
@@ -53,3 +53,16 @@ export const jsNumberForAddress = (address: string): number => {
   const seed = parseInt(addr, 16);
   return seed;
 };
+
+export const filterOperators = [
```
Comment (awrichar) [inline src/core/utils.ts:57]: Do we perhaps want some descriptive text with these? A few are obvious, but I don't even know what they all do.

File: src/modules/data/views/Messages/Messages.tsx [hunk 2]
```diff
@@ -57,17 +70,74 @@ export const Messages: () => JSX.Element = () => {
...
+    'created',
+    'group',
+    'hash',
+    'id',
+    'key',
+    'local',
+    'namespace',
+    'pending',
+    'pins',
+    'rejected',
```
Comment (awrichar) [inline src/modules/data/views/Messages/Messages.tsx:113]: local, pending, and rejected were rolled together as "state" in https://github.com/hyperledger/firefly/pull/296

**Review by awrichar — APPROVED** [review]
This is super slick 😎 Made a few minor comments, but I'm marking approved and will leave the rest up to you.

File: src/core/utils.ts (hunk 1 above)
Comment (shorsher) [reply src/core/utils.ts:57]: yeah I don't know either, it's a great point.

File: src/modules/data/views/Messages/Messages.tsx (hunk 2 above)
Comment (shorsher) [reply src/modules/data/views/Messages/Messages.tsx:113]: good catch, thanks

File: src/core/utils.ts (hunk 1 above)
Comment (awrichar) [reply src/core/utils.ts:57]: this appears to be the place: https://github.com/hyperledger/firefly/blob/04cd7184e0562a3a5a5344b0430bf68cc76415b1/internal/apiserver/restfilter.go#L126

**Pull request merged**